### PR TITLE
Added visual display pages for nist and neutron data

### LINKED ISSUE #8: Add visual fit benchmarking plots to tables
_From @Anders-Markvardsen on August 14, 2017 12:15_

When you run the python script in the attached file:

[runScriptsCopy.zip](https://github.com/mantidproject/mantid/files/1634117/runScriptsCopy.zip)

you get

image

for each row in this table, i.e. for each fitting problem, a start_for and Fit_for for the best fit on each row is returned. For example for the first row this is start_for_ENGINX_193749_calibration_spectrum_651_peak_19_1.pdf and Fit_for_ENGINX_193749_calibration_spectrum_651_peak_19_1.pdf. 

In this issue create one page (*.rst page which converted into *.html page) for each row, i.e. each fitting problem, which contains contains a picture of the start_for and fit_for images described above, and the following meta data:

* Which minimizer produced the best fit and the resulting chi-squared value
* The function name. For example 'name=LinearBackground;name=BackToBackExponential,I=597.076,A=1,B=0.05,X0=24027.5,S=22.9096'

Store these pages in a separate sub-folder, e.g. named 'IndividualResults'. Also move all related image file to this sub-folder.

The attached python script can be run with  run_data = "neutron", run_data = "nist" and run_data = "muon". Ensure that the individual-result page creation works for all these 3 cases. For the run_data = "nist" pages also add the meta data: link to the NIST url the fitting benchmarking tool is already generating to the page, e.g. in a 'see also section' at the bottom of the page.

It is anticipated that the rendering of *.pdf on html pages may not be straightforward. If e.g. this required important a new 3rd party abandon this approach. Instead create *.png versions of the *.pdf image files which are guaranteed to render on html pages.

One page is created for each fitting problem outputted both in *.rst and *.html. The former is needed to work with Mantid rst concept pages and the latter is needed when the fitting benchmarking tool is run separately. This is in line with how other outputs are produced by the fitting benchmarking tool.

Finally:

* update all links in the first column of the table shown above to point to the rst pages generated with this issue, and the corresponding generated html page contain a table where the hyperlinks works. Ensure this also works for run_data = "nist" and run_data = "muon". 

_Copied from original issue: mantidproject/mantid#20243_

### LINKED ISSUE #75: Visual display pages are broken on windows.
Tables do not point to the correct files on windows.

### PR DESCRIPTION
**Pull request by bb511**
#### Description of Work

Fixes #8 
Fixes #75
Visual display pages now show what was requested in the issue.

#### Testing Instructions

1. Run `example_runScript.py` in the mantidpython console.
2. Check if the links in the produced tables link to the right pages and if the pages display the correct things.

### CODE REVIEW

**Comment by bb511** [comment]
Note: this also fixes #75 .

**Comment by Anders-Markvardsen** [comment]
merge conflicts

File: fitbenchmarking/results_output.py [hunk 1]
```diff
@@ -153,42 +146,67 @@ def build_visual_display_page(prob_results, group_name, results_dir):
...
+        support_pages_dir = os.path.join(results_dir, "neutron", "tables", "support_pages")
+        fit_function_details_table = fit_rst_table_neutron(gb.function_def)
+        see_also_link = ''
 
     file_name = (group_name + '_' + problem_name).lower()
     file_path = os.path.join(support_pages_dir, file_name)
 
     rst_file_path = file_path.replace('\\', '/')
-    rst_link = "<file:///" + rst_file_path + "." + FILENAME_EXT_HTML + ">`__"
+    rst_link = "<file:///" +'' + rst_file_path + "." + FILENAME_EXT_HTML + ">`__"
```
Comment (Anders-Markvardsen) [inline fitbenchmarking/results_output.py]: why adding empty string here, i.e. ''?

File: fitbenchmarking/results_output.py [hunk 2]
```diff
@@ -153,42 +146,67 @@ def build_visual_display_page(prob_results, group_name, results_dir):
...
 
-    # Get path to the figures
     figures_dir = os.path.join(support_pages_dir, 'figures')
-
     figure_data = os.path.join(figures_dir, "Data_Plot_" + problem_name + "_1" + ".png")
     figure_fit = os.path.join(figures_dir, "Fit_for_" + problem_name + "_1" + ".png")
     figure_start = os.path.join(figures_dir, "start_for_" + problem_name + "_1" + ".png")
+    
+    if os.name == 'nt':
```
Comment (Anders-Markvardsen) [inline fitbenchmarking/results_output.py]: please add comment to why this is needed

**Comment by Anders-Markvardsen** [comment]
Get error 
```
---------------------------------------------------------------------------
UnboundLocalError                         Traceback (most recent call last)
C:\Midlertiddig\fitbenchmarking\example_scripts\example_runScripts.py in <module>()
     76                     use_errors=use_errors,
     77                     rst=True, save_to_file=True, color_scale=color_scale,
---> 78                     results_dir=results_dir)

C:\Midlertiddig\fitbenchmarking\fitbenchmarking\results_output.py in print_group_results_tables(minimizers, results_per_test, group_name, use_errors, simple_text, rst, save_to_file, color_scale, results_dir)
     79     """
     80     tables_dir = make_result_tables_directory(results_dir, group_name)
---> 81     linked_problems = build_indiv_linked_problems(results_per_test, group_name, results_dir)
     82
     83     accuracy_tbl, runtime_tbl = postproc.calc_accuracy_runtime_tbls(results_per_test, minimizers)

C:\Midlertiddig\fitbenchmarking\fitbenchmarking\results_output.py in build_indiv_linked_problems(results_per_test, group_name, results_dir)
    134         prev_name = name
    135         name_index = name + ' ' + str(prob_count)
--> 136         name = '`' + name + ' ' + build_visual_display_page(prob_results, group_name, results_dir)
    137         linked_problems.append(name)
    138

C:\Midlertiddig\fitbenchmarking\fitbenchmarking\results_output.py in build_visual_display_page(prob_results, group_name, results_dir)
    164
    165     file_name = (group_name + '_' + problem_name).lower()
--> 166     file_path = os.path.join(support_pages_dir, file_name)
    167
    168     rst_file_path = file_path.replace('\\', '/')

UnboundLocalError: local variable 'support_pages_dir' referenced before assignment
```

File: fitbenchmarking/results_output.py [hunk 3]
```diff
@@ -153,42 +146,67 @@ def build_visual_display_page(prob_results, group_name, results_dir):
...
     data_plot += ('-' * len(data_plot)) + '\n\n'
-    data_plot += '.. image:: ' + figure_data + '\n\n'
+    data_plot += '*Plot of the data considered in the problem*\n\n'
+    data_plot += ('.. image:: ' + figure_data + '\n' +
+                  '   :align: center' + '\n\n')
+
     starting_plot = 'Plot of the initial starting guess' + '\n'
     starting_plot += ('-' * len(starting_plot)) + '\n\n'
-    starting_plot += '.. figure:: ' + figure_start  + '\n\n'
+    starting_plot += '*Minimizer*: Levenberg-Marquardt \n\n'
```
Comment (Anders-Markvardsen) [inline fitbenchmarking/results_output.py]: Why Levenberg-Marquardt hardcoded here

**Comment by Anders-Markvardsen** [comment]
tested and works

**Pull request merged**